src/vampir/vampir.lisp

```diff
@@ -68,3 +68,256 @@
+                (mapcar (lambda (x) (expt 2 x))
+                        (alexandria:iota 32 :start 1))
+                (mapcar (lambda (x) (intern
+                                     (format nil "A~a" x) :keyword))
+                        (alexandria:iota 32)))
+               (list (make-equality :lhs (make-wire :var :a)
+                                    :rhs (in-plus 31)))
+               (list (make-tuples :wires (list
+                                          (in-tuple 31)
+                                          (make-wire :var :|()|)))))))))
+
+(defun range32 (a)
+  (make-application :func :range32
+                    :arguments (list a)))
+
+(defparameter *intrange32*
+  (make-alias :name :intRange32
+              :inputs (list :a)
+              :body (list (make-application :func :range32
+                                            :arguments (list (make-infix :op :+
+                                                                         :lhs (make-wire :var :a)
+                                                                         :rhs (make-constant :const 2147483648)))))))
+
+(defun intrange32 (a)
+  (make-application :func :intrange32
+                    :arguments (list a)))
+
+
+(defparameter *negative32*
+  (flet ((wires (num)
+           (make-wire :var (intern (format nil "A~a" num) :keyword))))
+    (make-alias
+     :name   :negative32
+     :inputs (list :a)
+     :body
+     (list (make-bind
+            :names (append (mapcar #'wires (alexandria:iota 32))
+                           (list (make-wire :var :|()|)))
+            :value (make-application
+                    :func      :intrange32
+                    :arguments (list (make-wire :var :a))))
+           (make-wire :var :a31)))))
+
+(defun negative32 (a)
+  (make-application :func :negative32
+                    :arguments (list a)))
+
+(defparameter *nonnegative32*
+  (make-alias :name :nonnegative32
+              :inputs (list :a)
+              :body (list
+                     (make-infix :op :-
+                                 :lhs (make-constant :const 1)
+                                 :rhs (make-application :func :negative32
+                                                        :arguments (list (make-wire :var :a)))))))
+(defun nonnegative32 (a)
+  (make-application :func :negative32
+                    :arguments (list a)))
+
+(defparameter *range31*
+  (labels ((in-plus (n)
+             (if (= n 0)
+                 (make-wire :var :a0)
+                 (make-infix :op :+
+                             :lhs (in-plus (- n 1))
+                             :rhs (make-infix :op :*
+                                              :lhs (make-constant :const (expt 2 n))
+                                              :rhs (make-wire :var (intern
+                                                                    (format nil "A~a" n) :keyword)))))))
```

Comment:
I see 

```lisp
(intern (format nil "A~a" n) :keyword)
```
everywhere, mind adding a function to the `geb.utils` package?

Maybe something like `prefix-keyword` that takes a string, and any other thing, and does this logic? 

call site would look like

```lisp
(prefix-keyword "A" 5)
```